plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -768,6 +768,12 @@ def merge_pkglists(self, other):
             # no collection with such name or no collection with such name and _pulp_repo_id
             else:
                 collections_to_add.append(new_collection)
+
+        # It is very important to call save() here to save recent modifications to the existing
+        # collections in the pkglist in the database before new collections will be added to
+        # the pkglist, because mongoengine does not allow to modify existing items in the list
+        # and add new items to the list at the same time.
+        self.save()
```

Comment:
Because of this change, I propose the function name change from `merge_pkglists` to `merge_pkglists_and_save`.

Also does it make sense to have an additional save() after L#777 so that the final state is also saved? I think it would be good but you would know better than I about that.